coalib/coala_modes.py

```diff
@@ -46,34 +46,17 @@ def mode_json(args, debug=False):
     import json
 
     from coalib.coala_main import run_coala
-    from coalib.misc.DictUtilities import inverse_dicts
-    from coalib.misc.Exceptions import get_exitcode
     from coalib.output.Logging import configure_json_logging
     from coalib.output.JSONEncoder import create_json_encoder
 
     if args.log_json:
         log_stream = configure_json_logging()
 
     JSONEncoder = create_json_encoder(use_relpath=args.relpath)
-    results = []
-
-    if args.show_bears:
-        try:
-            from coalib.parsing.FilterHelper import apply_filter
-
-            local_bears, global_bears = apply_filter(
-                'language', args.filter_by_language)
-            bears = inverse_dicts(local_bears, global_bears)
-            for bear, _ in sorted(bears.items(),
-                                  key=lambda bear_tuple:
-                                  bear_tuple[0].name):
-                results.append(bear)
-        except BaseException as exception:  # pylint: disable=broad-except
-            return get_exitcode(exception)
-    else:
-        results, exitcode, _ = run_coala(args=args, debug=debug)
 
-    retval = {'bears': results} if args.show_bears else {'results': results}
+    results, exitcode, _ = run_coala(args=args, debug=debug)
+
+    retval = {'results': results}
```

Comment:
do we change json output with that? If we have `show-bears` enabled, it's put into the `results` field in JSON, not into `bears` any more.

Reply:
No. If `show-bears` is specified along with json mode, execution flow won't come to this file. Have a look at https://github.com/coala/coala/pull/4726/files/fa62b7ba0d032b0d65a8c25c57f121b738a04202#diff-aed82cdd504caf613603b250c092f460L56. I've shifted this condition below `show-bears` condition in `coala.py` file.